monitor/reconfiguration-data-collector/src/main/java/eu/seaclouds/monitor/reconfigurationDataCollector/dataCollector/DeployerDC.java

```diff
@@ -0,0 +1,195 @@
+                    .getInternalComponentType();
+
+
+            while (true) {
+                  try {
+
+                        for (String metric : getApplicationMetrics()) {
+                              if (dcAgent.shouldMonitor(new InternalComponent(
+                                          monitoredTargetType, monitoredTargetId),
+                                          metric)) {
+                                
+                                    username = dcAgent.getParameters(metric)
+                                                .get("userName");
+                                    password = dcAgent.getParameters(metric)
+                                                .get("password");
+                                    deployerIp = dcAgent.getParameters(metric)
+                                                .get("deployerIp");
+                                    deployerPort = dcAgent.getParameters(metric)
+                                                .get("deployerPort");
+                                    
+                                    
+                                    BrooklynApi deployer=new BrooklynApi("http://"+deployerIp+":" + deployerPort + "/",
+                                            username, password);
+                                    
+                                    List<ApplicationSummary> apps = deployer.getApplicationApi().list(null);
+                                    
+                                    
+                                    
+                                    for(ApplicationSummary app: apps){
+                                    
+                                        Set<String> loc=app.getSpec().getLocations();
+                                        String providers="";
+                                        
+                                        for(String l:loc){
+                                           providers=providers+"-"+deployer.getLocationApi().get(l, "").getName();
+                                        }
+                                        
+                                        if(app.getId().equals(monitoredTargetType)){
+                                            dcAgent.send(new InternalComponent(monitoredTargetType,monitoredTargetId), metric,app.getId()+","+app.getStatus()+","+providers);         
+                                        }
+                                    }
+                         
+                                    Thread.sleep(Integer.parseInt(dcAgent.getParameters(
+                                                metric).get("samplingTime")) * 1000);
+                              }
+
+                        }
+
+                  } catch (InterruptedException e) {
+                   logger.debug("Data collector interrupeted while waiting to sample a new datum!");;
+                   e.printStackTrace();
```

Comment:
I've seen this pattern all over the PR. Please use proper the logger and remove `e.printStackTrace()` wherever possible. Also consider logger.error + throw RuntimeException where sensible